apps/builder/app/builder/features/topbar/builder-mode.tsx

```diff
@@ -69,6 +69,18 @@ export const BuilderModeDropDown = () => {
     setActiveMode(undefined);
   };
 
+  const getDefaultMessage = () => {
+    if (
+      isContentModeAllowed &&
+      isFeatureEnabled("contentEditableMode") &&
+      isDesignModeAllowed
+    ) {
+      return "Select Design or Content mode";
+    }
+
+    return !isDesignModeAllowed ? "Select Content mode" : "Select Design mode";
```

Comment:
try to avoid negations when possible

```suggestion
    return isDesignModeAllowed ? "Select Design mode" : "Select Content mode";
```

Reply:
overall this function could be expressed as a map, something like: 

```js
const messages = {
  both: "Select Design or Content mode",
  content: "Select Content mode",
  design: "Select Design mode" 
}

const mode = ContentModeAllowed &&
      isFeatureEnabled("contentEditableMode") &&
      isDesignModeAllowed ? "both" : isDesignModeAllowed ? 'design' : 'content'
```

Something like this, we tend to prefer state machines over "ifs"

Reply:
I am fine merging it though as-is, just a tip for the future

Reply:
> overall this function could be expressed as a map, something like:
> 
> ```js
> const messages = {
>   both: "Select Design or Content mode",
>   content: "Select Content mode",
>   design: "Select Design mode" 
> }
> 
> const mode = ContentModeAllowed &&
>       isFeatureEnabled("contentEditableMode") &&
>       isDesignModeAllowed ? "both" : isDesignModeAllowed ? 'design' : 'content'
> ```
> 
> Something like this, we tend to prefer state machines over "ifs", even though here you still have a conditional

thanks for the advice. I've come across state machines in automata theory, but haven't come across using that in code, is there any article you'd recommend reading.